streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/CheckpointProcessor.java

```diff
@@ -52,30 +66,29 @@ public class CheckpointProcessor {
 
   @Autowired private SavePointService savePointService;
 
-  public void process(Long appId, CheckPoints checkPoints) {
-    CheckPoints.Latest latest = checkPoints.getLatest();
-    if (latest == null || latest.getCompleted() == null) {
-      return;
-    }
-    CheckPoints.CheckPoint checkPoint = latest.getCompleted();
+  public void process(Long appId, String jobID, @Nonnull CheckPoints checkPoints) {
+    checkPoints.getLatestCheckpoint().forEach(checkPoint -> process(appId, jobID, checkPoint));
+  }
+
+  private void process(Long appId, String jobID, @Nonnull CheckPoints.CheckPoint checkPoint) {
     Application application = applicationService.getById(appId);
```

Comment:
This query can be removed, and we can pass some parameters from caller. It can reduce the pressure of DB.

For example, jobID, application.cpFailedTrigger(), etc.